src/components/tabs/js/tabsController.js

```diff
@@ -55,41 +66,70 @@ function MdTabsController ($scope, $element, $window, $timeout, $mdConstant, $md
    * Perform initialization for the controller, setup events and watcher(s)
    */
   function init () {
-    $scope.selectedIndex = $scope.selectedIndex || 0;
+    ctrl.selectedIndex = ctrl.selectedIndex || 0;
     compileTemplate();
-    configureWatchers();
     bindEvents();
     $mdTheming($element);
     $timeout(function () {
       updateHeightFromContent();
       adjustOffset();
       updatePagination();
-      ctrl.tabs[$scope.selectedIndex] && ctrl.tabs[$scope.selectedIndex].scope.select();
+      ctrl.tabs[ctrl.selectedIndex] && ctrl.tabs[ctrl.selectedIndex].scope.select();
       loaded = true;
     });
   }
 
+  /**
+   * Compiles the template provided by the user.  This is passed as an attribute from the tabs
+   * directive's template function.
+   */
   function compileTemplate () {
     var template = $attrs.$mdTabsTemplate,
         element  = angular.element(elements.data);
     element.html(template);
-    $compile(element.contents())(ctrl.parent);
+    $compile(element.contents())(ctrl.scope);
     delete $attrs.$mdTabsTemplate;
   }
 
+  /**
+   * Binds events used by the tabs component.
+   */
   function bindEvents () {
     angular.element($window).on('resize', handleWindowResize);
+    $scope.$on('$destroy', cleanup);
   }
 
-  function configureWatchers () {
-    $mdUtil.initOptionalProperties($scope, $attrs);
-    $attrs.$observe('mdNoBar', function (value) { $scope.noInkBar = angular.isDefined(value); });
-    $scope.$watch('selectedIndex', handleSelectedIndexChange);
-    $scope.$watch('dynamicHeight', function (value) {
-      if (value) $element.addClass('md-dynamic-height');
-      else $element.removeClass('md-dynamic-height');
-    });
-    $scope.$on('$destroy', cleanup);
+  /**
+   * Defines a two-way data-binding without requiring an isolated scope.
+   * @param key
+   * @param attr
+   * @param defaultValue
+   * @param handler
+   */
+  function defineBinding (key, attr, defaultValue, handler) {
+    var value = defaultValue, model;
+    defineProperty(key, function (newValue, oldValue) {
+      model && model.assign(ctrl.scope, newValue);
+      handler(newValue, oldValue);
+    }, value);
+    if ($attrs.hasOwnProperty(attr)) {
+      model = $parse($attrs[attr]);
+      $scope.$watch($attrs[attr], function (newValue) {
+        ctrl[key] = newValue || 0;
+      });
+    }
+  }
+
+  /**
+   * Defines boolean attributes with default value set to true.  (ie. md-stretch-tabs with no value
+   * will be treated as being truthy)
+   * @param key
+   * @param handler
+   */
+  function defineBooleanAttribute (key, handler) {
+    var attr = 'md' + key.charAt(0).toUpperCase() + key.substr(1);
```

Comment:
This is actually generating the attribute with the `md` prefix, not just normalizing.